### PR TITLE
Enable hooks by default for FB React Native renderer

### PR DESCRIPTION
**Pull request by bvaughn**

### CODE REVIEW

File: packages/shared/forks/ReactFeatureFlags.native-fb.js [hunk 1]
```diff
@@ -23,6 +22,7 @@ export const {
 } = require('ReactFeatureFlags');
```
Comment (sebmarkbage) [inline packages/shared/forks/ReactFeatureFlags.native-fb.js]: The rest of these, are they still relevant to be dynamic or can we just turn them on or off?

Comment (yungsters) [reply packages/shared/forks/ReactFeatureFlags.native-fb.js]: Only `debugRenderPhaseSideEffects` is currently dynamically configured internally.

We always set `debugRenderPhaseSideEffectsForStrictMode` to true.

Comment (bvaughn) [reply packages/shared/forks/ReactFeatureFlags.native-fb.js]: Looks like `debugRenderPhaseSideEffects ` is a dynamic value, `debugRenderPhaseSideEffectsForStrictMode` is being set to true in `InitializeReactFeatureFlags`, and nothing is changing the default value of `warnAboutDeprecatedLifecycles`.

**Comment by bvaughn** [comment]
I've pushed another commit to this branch that makes all but one of the flags (`debugRenderPhaseSideEffects`) static.

Here are the new flag values:

| flag | value | previous value? |
| :--- | :--- | :--- |
| `debugRenderPhaseSideEffects` | (dynamic) | (dynamic) |
| `debugRenderPhaseSideEffectsForStrictMode` | `true` |  `true` |
| `disableInputAttributeSyncing` | `false` | `false` |
| `enableHooks` | `true` | (dynamic) |
| `replayFailedUnitOfWorkWithInvokeGuardedCallback` | `__DEV__` | `false`  |
| `warnAboutDeprecatedLifecycles` | `true` | `true` |

**Pull request merged**